### ISSUE TITLE
Classes without annotations. Annotations without images

### ISSUE THREAD

**Issue by nmc-costa** [TAO-Dataset/tao#28]
Good Morning. 

Thank you so much for the hard work in doing this dataset. 
We have being using it in yolov5 to train object detection.

However, while the COCO Dataset has all the classes with annotations and all the annotations with images, the TAO dataset ,downloaded from https://motchallenge.net/tao_download.php, has classes without annotations, and annotations without images. 

If you want I can share an excel with the ones that we found.

Is it normal, or was it a problem during download?
How should we download to have the full dataset?

Thanks,
NC

**Comment by achalddave** [comment]
Hello, glad to hear you're finding the dataset useful! Sorry for this late reply.

You're right - there are classes without annotations in train+val. Some categories are present only in TAO test. For these categories, the expectation is that methods will learn detectors on LVIS v0.5 for such classes without video training data. There are also some LVIS classes which are not present in TAO; these are included in the "categories" field for consistency with the LVIS annotations, but are not used for evaluation.

I'm not sure I understand whta you mean by "annotations without images." This should not be the case. Have you downloaded the AVA and HACS subsets, as indicated on this page: https://motchallenge.net/tao_download.php ?
<img width="1117" alt="Screen Shot 2021-02-22 at 2 47 01 PM" src="https://user-images.githubusercontent.com/1168881/108761443-dcddf580-751c-11eb-81fe-e8f4403ab963.png">

**Comment by nmc-costa** [comment]
Thank you for the response,

Then, to have all the images and corresponding annotations the best method to download them all is:
wget "https://motchallenge.net/data/1-TAO_TRAIN.zip" 
wget "https://motchallenge.net/data/2-TAO_VAL.zip" 
wget "https://motchallenge.net/data/3-TAO_TEST.zip" 

then "Download TAO AVA and HACS videos", correct?

Thanks,

**Comment by nmc-costa** [comment]
> Hello, glad to hear you're finding the dataset useful! Sorry for this late reply.
> 
> You're right - there are classes without annotations in train+val. Some categories are present only in TAO test. For these categories, the expectation is that methods will learn detectors on LVIS v0.5 for such classes without video training data. There are also some LVIS classes which are not present in TAO; these are included in the "categories" field for consistency with the LVIS annotations, but are not used for evaluation.
> 
> I'm not sure I understand whta you mean by "annotations without images." This should not be the case. Have you downloaded the AVA and HACS subsets, as indicated on this page: https://motchallenge.net/tao_download.php ?
> <img alt="Screen Shot 2021-02-22 at 2 47 01 PM" width="1117" src="https://user-images.githubusercontent.com/1168881/108761443-dcddf580-751c-11eb-81fe-e8f4403ab963.png">

Additionally, @achalddave, to have a fulltao.json to test for missing classes, images and annotations in the downloaded dataset, which annotations should be merged: 

1. {train,validation}.json: Contains annotations for LVIS categories in TAO train and validation.
2. {train,validation}_with_freeform.json: Contains annotations for all categories in TAO train and validation.
3. test_without_annotations.json: JSON for test videos. The 'images' and 'videos' fields contain the images and videos that will be evaluated on the test set.
4. test_categories.json: Contains a list of categories which will be evaluated on the TAO test set.  

All of them,1+2+3+4. or only 1+2?

**Comment by achalddave** [comment]
> Thank you for the response,
> 
> Then, to have all the images and corresponding annotations the best method to download them all is:
> wget "https://motchallenge.net/data/1-TAO_TRAIN.zip"
> wget "https://motchallenge.net/data/2-TAO_VAL.zip"
> wget "https://motchallenge.net/data/3-TAO_TEST.zip"
> 
> then "Download TAO AVA and HACS videos", correct?

Yup! Sorry for the multi-part downloads; the dataset is large and this reduces issues where the download fails halfway.

> Additionally, @achalddave, to have a fulltao.json to test for missing classes, images and annotations in the downloaded dataset, which annotations should be merged:

Only (1) is needed for most purposes, such as evaluating on the test server at https://motchallenge.net/results/TAO_Challenge/. (2) can be used for evaluating open-world or user-initialized tracking -- it contains objects with free-form class annotations (e.g., objects not in the vocabulary, like "nerf gun", or parts/combinations of objects, like "part of cardboard box").

(3) is provided simply for ease of use - e.g., if you have code that expects a json file for the test set. (4) is provided so you can run your tracker on only a subset of classes which are evaluated on the test set.

Keep the questions coming! Happy to help.

**Comment by nmc-costa** [comment]
@achalddave Thank you, I will give it a try. Then. I will try to post here the results.

**Comment by achalddave** [comment]
(Closing due to inactivity, feel free to re-open)

**Comment by nmc-costa** [comment]
Sorry for the late reply. But I confirmed it, and after following the steps, there is no more missing images. However, as pointed in your main page, not all the categories have images associated. Thank you.
